sdk-java/src/main/java/io/littlehorse/sdk/worker/internal/PollThread.java

```diff
@@ -1,37 +1,33 @@
 package io.littlehorse.sdk.worker.internal;
 
-import io.grpc.stub.StreamObserver;
+import com.google.common.collect.Iterators;
 import io.littlehorse.sdk.common.proto.LittleHorseGrpc;
-import io.littlehorse.sdk.common.proto.PollTaskRequest;
-import io.littlehorse.sdk.common.proto.PollTaskResponse;
 import io.littlehorse.sdk.common.proto.TaskDefId;
 import io.littlehorse.sdk.worker.internal.util.VariableMapping;
 import java.io.Closeable;
 import java.lang.reflect.Method;
+import java.util.Iterator;
 import java.util.List;
 import java.util.concurrent.Semaphore;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
-public class PollThread extends Thread implements Closeable, StreamObserver<PollTaskResponse> {
+public class PollThread extends Thread implements Closeable {
 
-    private StreamObserver<PollTaskRequest> pollClient;
-    private final String taskWorkerId;
-    private final TaskDefId taskDefId;
-    private final String taskWorkerVersion;
-    private final Semaphore semaphore = new Semaphore(1);
+    private final Iterator<PollTaskStub> activePollClients;
+    private final Semaphore semaphore;
```

Comment:
Oh no, not the IntelliJ naming :joy: 

What does this `Semaphore` represent? Is it `availableInflightTasks`?